jolt-sdk/src/lib.rs

```diff
@@ -66,3 +68,54 @@ impl Proof {
     }
 }
 
+pub struct BumpAllocator {
+    offset: UnsafeCell<usize>,
+}
+ 
+unsafe impl Sync for BumpAllocator {}
+
+extern "C" {
+    static _HEAP_PTR: u8;
+}
+
+fn heap_start() -> usize {
+    unsafe {
+        _HEAP_PTR as *const u8 as usize
+    }
+}
+
+impl BumpAllocator {
+    pub const fn new() -> Self {
+        Self {
+            offset: UnsafeCell::new(0),
+        }
+    }
+
+    pub fn free_memory(&self) -> usize {
+        heap_start() + (self.offset.get() as usize)
+    }
+
+    pub fn fake_alloc(&self, layout: Layout) -> *mut u8 {
+        unsafe {
+            self.alloc(layout)
+        }
+    }
```

Comment:
what is this for?

Reply:
Oh my bad that's from something I was testing. Removed.